### ISSUE TITLE
GetOrCreateIndex implementation

### ISSUE THREAD

**Issue by themisir** [meilisearch/meilisearch-dotnet#82]
In the below implementation we're trying to create index with given name, if we get an exception that matches specific exception message we try getting index. Instead of that wouldn't it be good to first try getting index if the result is null then trying to create a new index? In that case we'll also decrease call/request counts in case of index is already exists.

https://github.com/meilisearch/meilisearch-dotnet/blob/72c548cf336b96800c5c5e406c69549e59998791/src/Meilisearch/MeilisearchClient.cs#L114-L131

My suggested implementation:

```csharp
private async Task<Index> GetOrCreateIndex(string uid)
{
  var index = await _client.GetIndex(uid);
  if (index == null)
  {
      index = await _client.CreateIndex(uid);
  }

  return index;
}
```

**Comment by curquiza** [comment]
The current implementation of `GetOrCreateIndex` is indeed not optimal for the moment. But we plan to change it soon, with some other functions. Indeed, now MeiliSearch is able to create an index when you add documents if the index does not exist yet, which is really convenient.

We opened an issue about the involved changes: meilisearch/integration-guides#48. There are sections about `getOrCreateIndex` that I will update accordingly soon! :wink:

**Comment by curquiza** [comment]
I close this issue, it has been implemented by @alallema in #118 😇
